scanpy/tools/_leiden.py

```diff
@@ -30,6 +30,7 @@ def leiden(
     use_weights: bool = True,
     n_iterations: int = -1,
     partition_type: Optional[Type[MutableVertexPartition]] = None,
+    neighbors_key: Optional[str] = None,
```

Comment:
What do you think about this being `connectivities`, with a default value of `"connectivity"` instead? The distances graph isn't used, and really the graph could come from anywhere.

This would also remove the need for `NeighborsView` here, and fits more with what I'm doing for `leiden_multiplex`.